src/pages/blogpostDetail/blogpostPage.jsx

```diff
@@ -0,0 +1,33 @@
+import {Link, useParams} from "react-router-dom";
+import posts from "/src/constants/data.json";
+import moment from "moment";
+
+function formatDate(dateString) {
+    const date = moment(dateString);
+    const formattedDate = date.format('DD MMMM YYYY');
+
+    return formattedDate;
+}
+
+function Blogpost() {
+    const {blogId} = useParams();
+
+
+    return (
+        <>
+        <h1>Blogpost Detail {blogId}</h1>
+            {posts.map((post) => {
```

Comment:
Je krijgt alle posts op de detail pagina omdat je hier door alle posts heen loopt. Dit is niet nodig,  je hebt hier de post die je wilt tonen namelijk al opgehaald door de filter method. De map method kan dus weggelaten worden.